torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -2137,6 +2159,9 @@ void TensorExprKernel::compile() {
     // since NNC views it as contiguous. Only convert it to the right
     // strides at the end of the kernel (if already contiguous it's a no-op)
     Tensor* properly_strided_output = convertOutputToCorrectStrides(output);
+    if (tensors_.at(output->unique()) != properly_strided_output) {
+      tensor_stmts.push_back(Stmt::clone(properly_strided_output->stmt()));
```

Comment:
Instead of having a vector of Stmts and cloning tensor stmts into it, why can't we have a Block and append tensor stmts to it immediately, without cloning anything?

Reply:
I ran into some issues when I tried without cloning the stmts. But I wasn't adding them to a Block then. Let me try this cleanup again in a follow up PR.